File: src/Features/CalculatorModel.cs
using Qml.Net;
using System;

namespace Features
{
    public class CalculatorModel
    {
        private bool _isValid = false;
        private string _computedResult = "";

        [NotifySignal]
        public string ComputedResult
        {
            get
            {
                return _computedResult;
            }
            set
            {
                if (_computedResult == value)
                {
                    // No signal is raised, Qml doesn't update any bound properties.
                    return;
                }

                _computedResult = value;
                this.ActivateSignal("computedResultChanged");
            }
        }

        [NotifySignal]
        public bool IsValid
        {
            get
            {
                return _isValid;
            }
            set
            {
                if (_isValid == value)
                {
                    // No signal is raised, Qml doesn't update any bound properties.
                    return;
                }

                _isValid = value;
                this.ActivateSignal("isValidChanged");
            }
        }

        public void Add(string inputValue1, string inputValue2)
        {
            ComputedResult = Convert.ToString(decimal.Parse(inputValue1) + decimal.Parse(inputValue2));
        }

        public void Subtract(string inputValue1, string inputValue2)
        {
            ComputedResult = Convert.ToString(decimal.Parse(inputValue1) - decimal.Parse(inputValue2));
        }

        public void Multiply(string inputValue1, string inputValue2)
        {
            ComputedResult = Convert.ToString(decimal.Parse(inputValue1) * decimal.Parse(inputValue2));
        }

        public void Divide(string inputValue1, string inputValue2)
        {
            var value1 = decimal.Parse(inputValue1);
            var value2 = decimal.Parse(inputValue2);

            if (value2 == 0)
            {
                ComputedResult = $"Cannot devide by zero.";
                return;
            }

            ComputedResult = Convert.ToString(value1 / value2);
        }
    }
}

File: src/Features.FSharp/CalculatorModel.fs
namespace Features

open Qml.Net

type CalculatorModel() =
    let mutable isValid = false
    let mutable computedResult = ""

    [<NotifySignal>]
    member this.ComputedResult
        with get() = computedResult
        and set(value) = 
            if value <> computedResult then
                computedResult <- value
                this.ActivateSignal("computedResultChanged") |> ignore

    [<NotifySignal>]
    member this.IsValid
        with get() = isValid
        and set(value) =
            if value <> isValid then
                isValid <- value
                this.ActivateSignal("isValidChanged")  |> ignore

    member this.Add(inputValue1: string, inputValue2: string) =
        let a = decimal inputValue1
        let b = decimal inputValue2
        let result = a + b
        this.ComputedResult <- string result

    member this.Subtract(inputValue1: string, inputValue2: string) =
        let a = decimal inputValue1
        let b = decimal inputValue2
        let result = a - b
        this.ComputedResult <- string result

    member this.Multiply(inputValue1: string, inputValue2: string) =
        let a = decimal inputValue1
        let b = decimal inputValue2
        let result = a * b
        this.ComputedResult <- string result

    member this.Divide(inputValue1: string, inputValue2: string) =
        let a = decimal inputValue1
        let b = decimal inputValue2
        try
            let result = a / b
            this.ComputedResult <- string result
        with | :? System.DivideByZeroException -> this.ComputedResult <- "Division by zero!"
